From 5d696b862116eeb994c24ba19a1bd331f3d7f00c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:44 -0400
Subject: [usb] option: add Olivetti Olicard 500

Message-id: <1410887463-20674-393-git-send-email-dzickus@redhat.com>
Patchwork-id: 95603
O-Subject: [RHEL7 PATCH 392/591] usb: option: add Olivetti Olicard 500
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 533b3994610f316e5cd61b56d0c4daa15c830f89
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Apr 25 18:49:18 2014 +0200

    usb: option: add Olivetti Olicard 500

    Device interface layout:
    0: ff/ff/ff - serial
    1: ff/ff/ff - serial AT+PPP
    2: 08/06/50 - storage
    3: ff/ff/ff - serial
    4: ff/ff/ff - QMI/wwan

    Cc: <stable@vger.kernel.org>
    Reported-by: Julio Araujo <julio.araujo@wllctel.com.br>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 367c7f0..6335222 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -349,6 +349,7 @@ static void option_instat_callback(struct urb *urb);
 #define OLIVETTI_PRODUCT_OLICARD100  0xc000
 #define OLIVETTI_PRODUCT_OLICARD145  0xc003
 #define OLIVETTI_PRODUCT_OLICARD200  0xc005
+#define OLIVETTI_PRODUCT_OLICARD500  0xc00b
 
 /* Celot products */
 #define CELOT_VENDOR_ID    0x211f
@@ -1545,6 +1546,9 @@ static const struct usb_device_id option_ids[] = {
  { USB_DEVICE(OLIVETTI_VENDOR_ID, OLIVETTI_PRODUCT_OLICARD200),
   .driver_info = (kernel_ulong_t)&net_intf6_blacklist
  },
+ { USB_DEVICE(OLIVETTI_VENDOR_ID, OLIVETTI_PRODUCT_OLICARD500),
+  .driver_info = (kernel_ulong_t)&net_intf4_blacklist
+ },
  { USB_DEVICE(CELOT_VENDOR_ID, CELOT_PRODUCT_CT680M) }, /* CT-650 CDMA 450 1xEVDO modem */
  { USB_DEVICE_AND_INTERFACE_INFO(SAMSUNG_VENDOR_ID, SAMSUNG_PRODUCT_GT_B3730, USB_CLASS_CDC_DATA, 0x00, 0x00) }, /* Samsung GT-B3730 LTE USB modem.*/
  { USB_DEVICE(YUGA_VENDOR_ID, YUGA_PRODUCT_CEM600) },
-- 
1.7.1